### PR TITLE
Adds ELSER V2 PS and PA benchmarks

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds benchmark info of ELSER V2 platform specific and platform agnostic to the documentation.

### Preview

[Benchmark info](https://stack-docs_2579.docs-preview.app.elstc.co/guide/en/machine-learning/master/elser-benchmarks.html)

### CODE REVIEW

File: docs/en/stack/ml/nlp/images/ml-nlp-elser-v2-ps-bm-results.png
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/images/ml-nlp-elser-v2-ps-bm-results.png:1]: (optional) Should we remove the dots on this chart to make it consistent with the other charts?

Comment (maxhniebergall) [reply docs/en/stack/ml/nlp/images/ml-nlp-elser-v2-ps-bm-results.png:1]: Is it important for the charts to have a consistent style and look?

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:469]: (optional) While these numbers are accurate, we probably don't want to imply a higher level of consistency than there is. I propose that we round these rates to include no decimal points: 26 doc/s and 14 doc/s respectively to represent the relatively high variability of these values. If you decide to do this rounding, there are a few other numbers below that should also be rounded.

I realize that I included this level of specificity in my report, but upon reflection I think it might not have been the best choice.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:478]: Since we are mentioning the number of allocations, we should specify that these tests were performed on a 16vCPU machine.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:486]: (optional) Do we need this section? It seems to mostly repeat the information that was in the summary. Perhaps they could be combined into one section?

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:411]: In this section, I think we should mention the following:
* ELSER V1 has worse throughput and worse retrieval performance.
* ELSER V1 embeddings are not compatible with ELSER V2 embeddings, thus migrating from ELSER V1 to ELSER V2 requires reindexing from source.
* We don't recommend ELSER V1 for new users of ELSER; ELSER V1 is only recommended for users who want to use semantic search and are unwilling or unable to upgrade to v8.11.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:393]: (optional) I think it would be good to mention that V2 ". . . is *only* designed to run on Linux . . ." just as this is mentioned below.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:400]: (optional) Perhaps it would be good to mention the performance improvements in V2 such as:
"The biggest change in ELSER V2 (other than the large improvements in performance) is the introduction of . . ."

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:414]: Because these models have multiple types of performance (inference throughput, inference and query latency, and embedding/semantic search quality), we should be clear that the only metric that we know has improved is throughput. Thus, we could word this as:

"However, ELSER V1 
in {current} has higher throughput than the initial version due to {es} platform improvements"

**Review by maxhniebergall — CHANGES_REQUESTED** [review]
Overall, I think this is a great distillation of my benchmarking report. There are a few small but important changes that I think we should make before merging this report. There were also a number of other changes that I proposed, but which I think mostly come down to personal preference - these comments were marked with (optional) and you should decide if you think its a worthwhile improvement.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:411]: As in the latest commit, all references to ELSER V1 are removed, I did not add these bits to this page. However, they are all documented in the ELSER conceptual page [here](https://www.elastic.co/guide/en/machine-learning/master/ml-nlp-elser.html#elser-v2) and [here](https://www.elastic.co/guide/en/machine-learning/master/ml-nlp-elser.html#upgrade-elser-v2).

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:414]: Section has been deleted based on our conversation earlier.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:458]: lets add units to our performance numbers here: "22 docs/s", "26 docs/s".

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc
Comment (maxhniebergall) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:450]: Lets add the number of tokens per document here:

"These tests were performed on a 16vCPU machine, with all documents containing exactly 256 tokens."

Comment (maxhniebergall) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:450]: We might want to add another sentence like "The length of the documents in your dataset will have a significant impact on your throughput numbers." or similar

**Review by maxhniebergall — APPROVED** [review]
LGTM

**Pull request merged**
